.github/workflows/vulnerability-scanning-on-repo.yml

```diff
@@ -0,0 +1,125 @@
+      - uses: actions/checkout@v4
+      - name: Get version from package.json
+        uses: martinbeentjes/npm-get-version-action@main
+        id: package-version
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+      - name: Get node version
+        id: node
+        run: |
+          echo "version=$(node -v)" >> $GITHUB_OUTPUT
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+      - run: |
+          yarn install
+          yarn global add @cyclonedx/cyclonedx-npm
+          cyclonedx-npm --output-file ${{ env.BOM_FILE_PATH }}
+
+      - name: Upload SBOM artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ github.run_number }}-sbom.json
+          path: ${{ env.BOM_FILE_PATH }}
+  run_low_risk_vulnerability_scanner:
+    needs:
+      - generate_bom
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact       
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ github.run_number }}-sbom.json
+      - name: Scan SBOM
+        uses: aquasecurity/trivy-action@master
+        with:
+          exit-code: '0'
+          severity: 'UNKNOWN,LOW,MEDIUM'
+          ignore-unfixed: true
+          # https://github.com/aquasecurity/trivy-action/pull/314
+          scan-ref: ${{ env.BOM_FILE_PATH }}
+          scan-type: 'sbom'
+          scanners: 'vuln'
+  run_high_risk_vulnerability_scanner:
+    needs:
+      - generate_bom
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact       
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ github.run_number }}-sbom.json
+      - name: 
+        uses: aquasecurity/trivy-action@master
+        with:
+          exit-code: '1'
+          severity: 'HIGH,CRITICAL'
+          ignore-unfixed: true
+          # https://github.com/aquasecurity/trivy-action/pull/314
+          scan-ref: ${{ env.BOM_FILE_PATH }}
+          scan-type: 'sbom'
+          scanners: 'vuln'
+  write_sbom_to_repo:
+    needs:
+      - run_low_risk_vulnerability_scanner
+      - run_high_risk_vulnerability_scanner
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal access token.
+          fetch-depth: 0 # otherwise, there would be errors pushing refs to the destination repository.
+      - name: Download artifact       
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ github.run_number }}-sbom.json
+      - name: commit and push SBOM
```

Comment:
Not sure there's necessarily value in committing the SBOM to git as it's going to continuously change over time as your dependencies change e.g. via Dependabot PRs

I guess in JS world there's no way to publish the SBOMs alongside a release as we can in Java land?

Reply:
You're right, what I've done instead is remove the task that mutates the repo.

Instead the Dockerfile on the calling repo expects there to be an sbom that it can copy in. I have updated the `docker-push-to-registries.yml` workflow to pull down the artifact